Bug: KubeEleven panics after removing control nodes from k8s cluster

<!--
Note: Please search to see if an issue already exists for the bug you encountered.
-->

### Current Behavior:
<!-- A concise description of what you're experiencing. -->
After removing control nodes from a k8s cluster `kube-eleven` panics.
### Expected Behavior:
<!-- A concise description of what you expected to happen. -->
A k8s cluster without control nodes doesn't makes sense, this shouldn't pass through the validation phase of the config or this case should be handled appropriately by not `panic`-ing
### Steps To Reproduce:
<!--
Example: steps to reproduce the behavior:
1. In this environment...
2. With this config...
3. Run '...'
4. See error...
-->

1. apply the following config
```yaml
kubernetes:
  clusters:
    - name: k8s-cluster
      version: v1.21.0
      network: 192.168.2.0/24
      pools:
        control:
          - hetzner-control
        compute:
          - hetzner-compute
```
2. Apply new config with removed control nodes
```yaml
kubernetes:
  clusters:
    - name: k8s-cluster
      version: v1.21.0
      network: 192.168.2.0/24
      pools:
        control:
        compute:
          - hetzner-compute
```
3. Wait till the workflow reaches `kube-eleven` and see the panic message
```bash
panic: runtime error: index out of range [0] with length 0

goroutine 13 [running]:
github.com/Berops/claudie/services/kube-eleven/server/kube-eleven.(*KubeEleven).getClusterNodes(0x2c?)
        /Users/despire/go/src/github.com/berops/claudie/services/kube-eleven/server/kube-eleven/kube-eleven.go:160 +0x2b6
github.com/Berops/claudie/services/kube-eleven/server/kube-eleven.(*KubeEleven).generateTemplateData(0xc000333f38)
        /Users/despire/go/src/github.com/berops/claudie/services/kube-eleven/server/kube-eleven/kube-eleven.go:112 +0xff
github.com/Berops/claudie/services/kube-eleven/server/kube-eleven.(*KubeEleven).generateFiles(0xc000333f38)
        /Users/despire/go/src/github.com/berops/claudie/services/kube-eleven/server/kube-eleven/kube-eleven.go:87 +0xf3
github.com/Berops/claudie/services/kube-eleven/server/kube-eleven.(*KubeEleven).BuildCluster(0xc000333f38)
        /Users/despire/go/src/github.com/berops/claudie/services/kube-eleven/server/kube-eleven/kube-eleven.go:46 +0x165
main.(*server).BuildCluster.func1.1()
        /Users/despire/go/src/github.com/berops/claudie/services/kube-eleven/server/server.go:42 +0x66
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /Users/despire/go/pkg/mod/golang.org/x/sync@v0.0.0-20220819030929-7fc1605a5dde/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
        /Users/despire/go/pkg/mod/golang.org/x/sync@v0.0.0-20220819030929-7fc1605a5dde/errgroup/errgroup.go:72 +0xa5
exit status 2
make: *** [kubeEleven] Error 1
```

--- a/internal/manifest/manifest.go
+++ b/internal/manifest/manifest.go
@@ -103,11 +103,11 @@ type Cluster struct {
 	Name    string `validate:"required" yaml:"name"`
 	Version string `validate:"required,ver" yaml:"version"`
 	Network string `validate:"required,cidrv4" yaml:"network"`
-	Pools   Pool   `yaml:"pools"`
+	Pools   Pool   `validate:"dive" yaml:"pools"`
 }
 
 type Pool struct {
-	Control []string `yaml:"control"`
+	Control []string `validate:"min=1" yaml:"control"`
 	Compute []string `yaml:"compute"`
 }
 

--- a/manifests/claudie/kustomization.yaml
+++ b/manifests/claudie/kustomization.yaml
@@ -21,18 +21,18 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: claudieio/ansibler
-  newTag: 2962f5f-918
+  newTag: 80b4c06-1014
 - name: claudieio/builder
-  newTag: ee54904-895
+  newTag: 80b4c06-1014
 - name: claudieio/context-box
-  newTag: 0fd2a6c-896
+  newTag: 80b4c06-1014
 - name: claudieio/frontend
-  newTag: d0fceb4-911
+  newTag: 80b4c06-1014
 - name: claudieio/kube-eleven
-  newTag: a8096f9-928
+  newTag: 80b4c06-1014
 - name: claudieio/kuber
-  newTag: 681edde-931
+  newTag: 80b4c06-1014
 - name: claudieio/scheduler
-  newTag: 37fd718-924
+  newTag: 80b4c06-1014
 - name: claudieio/terraformer
-  newTag: f35e130-888
+  newTag: 80b4c06-1014

--- a/manifests/claudie/kustomization.yaml
+++ b/manifests/claudie/kustomization.yaml
@@ -21,18 +21,18 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: claudieio/ansibler
-  newTag: 80b4c06-1014
+  newTag: 78488f9-1016
 - name: claudieio/builder
-  newTag: 80b4c06-1014
+  newTag: 78488f9-1016
 - name: claudieio/context-box
-  newTag: 80b4c06-1014
+  newTag: 78488f9-1016
 - name: claudieio/frontend
-  newTag: 80b4c06-1014
+  newTag: 78488f9-1016
 - name: claudieio/kube-eleven
-  newTag: 80b4c06-1014
+  newTag: 78488f9-1016
 - name: claudieio/kuber
-  newTag: 80b4c06-1014
+  newTag: 78488f9-1016
 - name: claudieio/scheduler
-  newTag: 80b4c06-1014
+  newTag: 78488f9-1016
 - name: claudieio/terraformer
-  newTag: 80b4c06-1014
+  newTag: 78488f9-1016

--- a/manifests/claudie/kustomization.yaml
+++ b/manifests/claudie/kustomization.yaml
@@ -21,18 +21,18 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: claudieio/ansibler
-  newTag: f38664f-982
+  newTag: b978b3f-1037
 - name: claudieio/builder
-  newTag: ee54904-895
+  newTag: b978b3f-1037
 - name: claudieio/context-box
-  newTag: 0fd2a6c-896
+  newTag: b978b3f-1037
 - name: claudieio/frontend
-  newTag: d0fceb4-911
+  newTag: b978b3f-1037
 - name: claudieio/kube-eleven
-  newTag: a8096f9-928
+  newTag: b978b3f-1037
 - name: claudieio/kuber
-  newTag: 681edde-931
+  newTag: b978b3f-1037
 - name: claudieio/scheduler
-  newTag: 37fd718-924
+  newTag: b978b3f-1037
 - name: claudieio/terraformer
-  newTag: f35e130-888
+  newTag: b978b3f-1037

--- a/manifests/claudie/kustomization.yaml
+++ b/manifests/claudie/kustomization.yaml
@@ -21,18 +21,18 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: claudieio/ansibler
-  newTag: 78488f9-1016
+  newTag: e5d83f6-1017
 - name: claudieio/builder
-  newTag: 78488f9-1016
+  newTag: e5d83f6-1017
 - name: claudieio/context-box
-  newTag: 78488f9-1016
+  newTag: e5d83f6-1017
 - name: claudieio/frontend
-  newTag: 78488f9-1016
+  newTag: e5d83f6-1017
 - name: claudieio/kube-eleven
-  newTag: 78488f9-1016
+  newTag: e5d83f6-1017
 - name: claudieio/kuber
-  newTag: 78488f9-1016
+  newTag: e5d83f6-1017
 - name: claudieio/scheduler
-  newTag: 78488f9-1016
+  newTag: e5d83f6-1017
 - name: claudieio/terraformer
-  newTag: 78488f9-1016
+  newTag: e5d83f6-1017

--- a/manifests/testing-framework/kustomization.yaml
+++ b/manifests/testing-framework/kustomization.yaml
@@ -22,4 +22,4 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: claudieio/testing-framework
-  newTag: bc4393d-933
+  newTag: 80b4c06-1014

--- a/manifests/testing-framework/kustomization.yaml
+++ b/manifests/testing-framework/kustomization.yaml
@@ -22,4 +22,4 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: claudieio/testing-framework
-  newTag: 80b4c06-1014
+  newTag: 78488f9-1016

--- a/manifests/testing-framework/kustomization.yaml
+++ b/manifests/testing-framework/kustomization.yaml
@@ -22,4 +22,4 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: claudieio/testing-framework
-  newTag: 78488f9-1016
+  newTag: e5d83f6-1017

--- a/manifests/testing-framework/kustomization.yaml
+++ b/manifests/testing-framework/kustomization.yaml
@@ -22,4 +22,4 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: claudieio/testing-framework
-  newTag: e5d83f6-1017
+  newTag: b978b3f-1037

--- a/services/testing-framework/longhorn.go
+++ b/services/testing-framework/longhorn.go
@@ -11,7 +11,7 @@ import (
 )
 
 const (
-	maxLonghornCheck = 20 * 60 // max allowed time for pods of longhorn-system to be ready [seconds]
+	maxLonghornCheck = 30 * 60 // max allowed time for pods of longhorn-system to be ready [seconds]
 	sleepSecPods     = 20      // seconds for one cycle of longhorn checks (the node and pod checks)
 )
 

--- a/services/testing-framework/longhorn.go
+++ b/services/testing-framework/longhorn.go
@@ -11,7 +11,7 @@ import (
 )
 
 const (
-	maxLonghornCheck = 10 * 60 // max allowed time for pods of longhorn-system to be ready [seconds]
+	maxLonghornCheck = 20 * 60 // max allowed time for pods of longhorn-system to be ready [seconds]
 	sleepSecPods     = 20      // seconds for one cycle of longhorn checks (the node and pod checks)
 )
 

